src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/MarkAttributesWithAttributeUsageTests.cs

```diff
@@ -1,30 +1,81 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Testing;
 using Test.Utilities;
 using Xunit;
 using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
     Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.MarkAttributesWithAttributeUsageAnalyzer,
-    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.MarkAttributesWithAttributeUsageFixer>;
 using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
     Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.MarkAttributesWithAttributeUsageAnalyzer,
-    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.MarkAttributesWithAttributeUsageFixer>;
 
 namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
 {
     public partial class MarkAttributesWithAttributeUsageTests
     {
-        [Fact]
-        public async Task TestCSSimpleAttributeClass()
+        [Theory]
+        [InlineData(AttributeTargets.All)]
+        [InlineData(AttributeTargets.Assembly)]
+        [InlineData(AttributeTargets.Class)]
+        [InlineData(AttributeTargets.Constructor)]
+        [InlineData(AttributeTargets.Delegate)]
+        [InlineData(AttributeTargets.Enum)]
+        [InlineData(AttributeTargets.Event)]
+        [InlineData(AttributeTargets.Field)]
+        [InlineData(AttributeTargets.GenericParameter)]
+        [InlineData(AttributeTargets.Interface)]
+        [InlineData(AttributeTargets.Method)]
+        [InlineData(AttributeTargets.Module)]
+        [InlineData(AttributeTargets.Parameter)]
+        [InlineData(AttributeTargets.Property)]
+        [InlineData(AttributeTargets.ReturnValue)]
+        [InlineData(AttributeTargets.Struct)]
+        public async Task TestSimpleAttributeClass(AttributeTargets attributeTarget)
         {
-            await VerifyCS.VerifyAnalyzerAsync(@"
+            var attributeTargetsValue = "AttributeTargets." + attributeTarget.ToString();
+
+            await new VerifyCS.Test
+            {
```

Comment:
I think you also need to provide a `CodeActionIndex` value here based on the index of the `attributeTarget` field into the enumerated members of `AttributeTargets`

Reply:
I have tried a combination of both but it's only working for the `All` value. Basically anything else but `CodeActionIndex = 0` results in no codefix. Any value for the equivalency key gives no codefix and if I combine both properties then I have errors saying that the combination isn't good.